From 4dff1973116597210cf9fb1c102e5c9433d13b1d Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Sat, 14 Sep 2024 13:22:05 +0000
Subject: [PATCH] startkde: Fix wayland session restore saving

When we added manual saving support to plasma-shutdown a guard was
changed in the normal shutdown path. This guard was wrong, we want to
check we're in the restorePreviousLogout path here.

This amends 804976c5ecec1fbf5f6e7e09970a8269bdf748d2

BUG: 436318
---
 startkde/plasma-shutdown/shutdown.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/startkde/plasma-shutdown/shutdown.cpp b/startkde/plasma-shutdown/shutdown.cpp
index 11c002bcd80..ab93cca6ef7 100644
--- a/startkde/plasma-shutdown/shutdown.cpp
+++ b/startkde/plasma-shutdown/shutdown.cpp
@@ -84,7 +84,7 @@ void Shutdown::ksmServerComplete()
 {
     const KConfigGroup c(KSharedConfig::openConfig(QStringLiteral("ksmserverrc")), QStringLiteral("General"));
 
-    if (c.readEntry("loginMode", "restorePreviousLogout") == QLatin1String("restoreSavedSession")) {
+    if (c.readEntry("loginMode", "restorePreviousLogout") == QLatin1String("restorePreviousLogout")) {
         // Now record windows that are not session managed
         int ret = QProcess::execute(QStringLiteral(PLASMA_FALLBACK_SESSION_SAVE_BIN));
         if (ret) {
-- 
GitLab

